src/orca/services/nextflowtower/client.py

```diff
@@ -131,56 +125,205 @@ def get(self, path: str, **kwargs) -> dict[str, Any]:
         Returns:
             A dictionary from deserializing the JSON response.
         """
-        return self.request_json("GET", path, **kwargs)
+        response = self.request_json("GET", path, **kwargs)
+        if "totalSize" in response:
+            total_size = response.pop("totalSize")
+            key, items = response.popitem()
+            kwargs["params"] = kwargs["params"] or dict()
+            kwargs["params"]["offset"] = len(items)
+            remainder = self.request_paged("GET", path, **kwargs)
+            items.extend(remainder)
+            if len(items) != total_size:
+                message = f"Expected {total_size} items, but got: {items}"
+                raise HTTPError(message)
+            response[key] = items
+            response["totalSize"] = total_size
+        return response
+
+    def post(self, path: str, **kwargs) -> dict[str, Any]:
+        """Send an auth'ed POST request and parse the JSON response.
 
-    def get_user_info(self) -> dict[str, Any]:
-        """Describe current user.
+        See ``TowerClient.request`` for argument definitions.
 
-        Raises:
-            HTTPError: If the response lacks the expected keys.
+        Returns:
+            A dictionary from deserializing the JSON response.
+        """
+        return self.request_json("POST", path, **kwargs)
+
+    def unwrap(self, response: dict[str, Any], key: str) -> Any:
```

Comment:
Might it make sense to have functions like this one in a cross-service `utils` file? I can imagine this being useful for any service interacting with an API.

Reply:
I agree. Once we add a second service where we need to implement a client (like Tower), we can abstract the common functionality to a base client class. Fortunately, most services that we do plan on using have clients (_e.g._ Synapse, SevenBridges). We might need to reuse this code once we implement support for GA4GH WES, but I don't think that's prioritized for the time being.